src/servant/tt_002/3678.c

```diff
@@ -388,9 +392,189 @@ INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_80174F0C);
 
 INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_801753E4);
 
-INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_80175730);
+// This is a dupe of func_us_80175A78 with a slightly different offset
+// for the create entity params
+void func_us_80175730(Entity* self) {
+    const int paramOffset = 1;
+
+    s_TargetLocOffset_calc = -0x18;
+    if (!PLAYER.facingLeft) {
+        s_TargetLocOffset_calc = -s_TargetLocOffset_calc;
+    }
+    s_TargetLocationX = PLAYER.posX.i.hi + s_TargetLocOffset_calc;
+    s_TargetLocationY = PLAYER.posY.i.hi - 0x18;
+    switch (self->step) {
+    case 0:
+        func_us_801739D0(self);
+        func_us_80173994(self, 0xE);
+        break;
+
+    case 1:
+        s_AngleToTarget =
+            CalculateAngleToEntity(self, s_TargetLocationX, s_TargetLocationY);
+        s_AllowedAngle = GetTargetPositionWithDistanceBuffer(
+            s_AngleToTarget, self->ext.faerie.targetAngle, 0x180);
+        self->ext.faerie.targetAngle = s_AllowedAngle;
+        self->velocityY = -(rsin(s_AllowedAngle) << 5);
+        self->velocityX = rcos(s_AllowedAngle) << 5;
+        func_us_80173BD0(self);
+        self->posX.val += self->velocityX;
+        self->posY.val += self->velocityY;
+        s_DistToTargetLocation =
+            CalculateDistance(self, s_TargetLocationX, s_TargetLocationY);
+        if (s_DistToTargetLocation < 2) {
+            self->step++;
+        }
+        break;
+    case 2:
+        if (!D_80097A1A[paramOffset]) {
+            func_us_80173994(self, 0x14);
+            self->step = 5;
+            break;
+        }
+        if (SearchForEntityInRange(1, 0x27)) {
+            self->entityId = ENTITY_ID_SERVANT;
+            self->step = 0;
+            return;
+        }
+        func_us_80173994(self, 0x12);
+        if (s_ServantId == 6) {
```

Comment:
`FAM_ACTIVE_YOUSEI`

this probably can get a lint rule added.

Reply:
This is fixed, but I'll leave it open for trackability of the lint suggestion